flink-cdc-common/src/main/java/org/apache/flink/cdc/common/sink/DefaultHashFunctionProvider.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.cdc.common.sink;
+
+import org.apache.flink.cdc.common.data.RecordData;
+import org.apache.flink.cdc.common.event.DataChangeEvent;
+import org.apache.flink.cdc.common.event.OperationType;
+import org.apache.flink.cdc.common.event.TableId;
+import org.apache.flink.cdc.common.schema.Schema;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+/** the default implementation of hash function. */
+public class DefaultHashFunctionProvider implements HashFunctionProvider {
+    private static final long serialVersionUID = 1L;
+
+    @Override
+    public HashFunction getHashFunction(TableId tableId, Schema schema) {
+        return new DefaultHashFunction(schema);
+    }
+
+    static class DefaultHashFunction implements HashFunction {
+        private final List<RecordData.FieldGetter> primaryKeyGetters;
+
+        public DefaultHashFunction(Schema schema) {
+            primaryKeyGetters = createFieldGetters(schema);
+        }
+
+        @Override
+        public Integer apply(DataChangeEvent event) {
+            List<Object> objectsToHash = new ArrayList<>();
+            // Table ID
+            TableId tableId = event.tableId();
+            Optional.ofNullable(tableId.getNamespace()).ifPresent(objectsToHash::add);
+            Optional.ofNullable(tableId.getSchemaName()).ifPresent(objectsToHash::add);
+            objectsToHash.add(tableId.getTableName());
+
+            // Primary key
```

Comment:
This comment seems misleading.

Reply:
Is "Primary Key" causing missleading?
why?

Reply:
Never mind. Was wondering if something like "Extract Table ID" / "Extract primary key columns" would be better, but I think current comments are clear enough.